.pre-commit-config.yaml

```diff
@@ -0,0 +1,32 @@
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v5.0.0
+    hooks:
+      - id: end-of-file-fixer
+      - id: trailing-whitespace
+      - id: check-yaml
+      - id: check-json
+      - id: pretty-format-json
+        args: [ --autofix ]
+        exclude: frontend/package.*.json
+      - id: check-merge-conflict
+      - id: check-symlinks
+      - id: detect-private-key
+      - id: requirements-txt-fixer
+        files: requirements.*.in
+
+  - repo: https://github.com/crate-ci/typos
+    rev: v1.26.8
+    hooks:
+      - id: typos
+
+  # Ruff linter and formatter pre-commit
+  - repo: https://github.com/astral-sh/ruff-pre-commit.git
+    # Ruff version.
+    rev: v0.7.1
+    hooks:
+      # Run the linter.
+      - id: ruff
+        args: [ --fix ]
+      # Run the formatter.
+      - id: ruff-format
```

Comment:
@rodrigo-pessoa-lrn we actually have another story for `ruff` just an FYI so we can keep this PR a bit smaller and more straight forward, to just automated fixes mostly

https://learnosity.atlassian.net/browse/AIL-163